daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/RuntimeData.scala

```diff
@@ -862,88 +786,68 @@ sealed abstract class ModelGroupRuntimeData(
   final override def isArray = false
   override def namespaces: NamespaceBinding = ci.namespaces
 
-  lazy val variableMap = variableMapArg
-  lazy val encInfo = encInfoArg
-  lazy val ci = ciArg
-  lazy val groupMembers = groupMembersArg
-
-  override def preSerialization: Unit = {
-    super.preSerialization
-    variableMap
-    encInfo
-    ci
-    groupMembers
-  }
   @throws(classOf[java.io.IOException])
```

Comment:
I'm not really sure. I know Java uses reflection to find/call the writeObject method, and I think it explicitly looks a private writeObject method. Since it's private, I guess it can't really be inherited so maybe the reflection can't find it? Or maybe when Java does the reflection is just simply ignored any inherited functions and only looks at those explicitly defined on the class?

Or maybe this is all wrong and could work? I'm not sure.

Reply:
As an addendum to a previous comment, I think we can complete remove all transient params, as well as all the preserilize stuff.

The only place that really needs special serializtion logic is delay. We just need to add a writeObject for the Delay so that it forces evaluation of value and calls super.writeObject. That will set box to null, so the pass by name stuff isn't serialized. And the value is evaluated, so will be serialized correctly.

So if Delay parameters aren't made transient, and we add a little extra logic to Delay, all of this preserialization stuff goes away. I think...